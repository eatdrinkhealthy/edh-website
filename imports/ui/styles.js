import { css } from "glamor";

//
// Component Style Pattern (use with glamor)
//
// const componentRules = {
//    rule1: "value",
//      ...
//    ruleN: "value",
// };
// export const componentStyle = {
//    rules: componentRules,
//    className: `${css(componentRules)}`,
// };
//

const appLinksHeaderRules = {
  marginTop: "90px",
  marginLeft: "40px",
  fontSize: "16px",
  color: "darkbrown",
};
export const appLinksHeaderStyle = {
  rules: appLinksHeaderRules,
  className: `${css(appLinksHeaderRules)}`,
};

const appLinksButtonsRules = {
  marginTop: "14px",
  marginBottom: "20px",
  marginLeft: "40px",
};
export const appLinksButtonsStyle = {
  rules: appLinksButtonsRules,
  className: `${css(appLinksButtonsRules)}`,
};

const linkButtonRules = {
  backgroundColor: "darkslategray",
  border: "none",
  borderRadius: "4px",
  color: "white",
  padding: "8px 32px 8px 32px",
  marginRight: "8px",
  textAlign: "center",
  textDecoration: "none",
  fontSize: "12px",
};
export const linkButtonStyle = {
  rules: linkButtonRules,
  className: `${css(linkButtonRules)}`,
};
