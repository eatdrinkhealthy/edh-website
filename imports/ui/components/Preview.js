import React from "react";
import { appLinksHeaderStyle } from "../styles";
import LinkButton from "./LinkButton";

const Preview = () => (
  <div>
    <div className={`${appLinksHeaderStyle.className}`}>Early Version Preview...</div>
    <LinkButton text="Web App" url="https://app.eatdrinkhealthy.co/" />
    <LinkButton text="App Store" url="https://itunes.apple.com/app/id1116167404" />
  </div>
);

export default Preview;
