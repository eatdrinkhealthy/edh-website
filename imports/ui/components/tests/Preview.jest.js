/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import Preview from "../Preview";
import {
  formattedJSON,
  getAllComponentStyle,
} from "../../../utils/tests/testUtils";
import * as styles from "../../../ui/styles";

describe("<Preview />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<Preview />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches style / css snapshot", function () {
    const allStyle = getAllComponentStyle(<Preview />, styles);
    expect(formattedJSON(allStyle)).toMatchSnapshot();
  });
});
