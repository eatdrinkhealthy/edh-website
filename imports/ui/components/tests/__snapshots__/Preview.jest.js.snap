exports[`<Preview /> matches render snapshot 1`] = `
<div>
  <div
    className="css-m58pgh">
    Early Version Preview...
  </div>
  <a
    className="css-15br13v"
    href="https://app.eatdrinkhealthy.co/">
    Web App
  </a>
  <a
    className="css-15br13v"
    href="https://itunes.apple.com/app/id1116167404">
    App Store
  </a>
</div>
`;

exports[`<Preview /> matches style / css snapshot 1`] = `
"[
  {
    \"rules\": {
      \"marginTop\": \"90px\",
      \"marginLeft\": \"40px\",
      \"fontSize\": \"16px\",
      \"color\": \"white\"
    },
    \"className\": \"css-m58pgh\"
  },
  {
    \"rules\": {
      \"backgroundColor\": \"darkslategray\",
      \"border\": \"none\",
      \"borderRadius\": \"4px\",
      \"color\": \"white\",
      \"padding\": \"8px 32px 8px 32px\",
      \"textAlign\": \"center\",
      \"textDecoration\": \"none\",
      \"fontSize\": \"12px\"
    },
    \"className\": \"css-15br13v\"
  }
]"
`;
